pennylane/transforms/decompositions/clifford_plus_t/cliffordt_transform.py

```diff
@@ -0,0 +1,368 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transform function for the Clifford+T decomposition."""
+
+import math
+from itertools import product
+from typing import Sequence, Callable
+import pennylane as qml
+from pennylane.transforms.core import transform
+from pennylane.tape import QuantumTape
+from pennylane.transforms.optimization import (
+    cancel_inverses,
+    commute_controlled,
+    merge_rotations,
+    remove_barrier,
+)
+from pennylane.transforms.optimization.optimization_utils import find_next_gate, _fuse_global_phases
```

Comment:
Since we are needing to access a `_fuse_global_phases` from another file, I wonder if we should either make a public version of this function or duplicate the code here so that we don't end up with tight coupling and unexpected dependencies.

Reply:
I don't mind it making public, it's a handy function to have whenever the tapes have global phase operations.